examples/examples/middleware.rs

```diff
@@ -46,43 +52,71 @@ async fn main() -> anyhow::Result<()> {
 		.try_init()
 		.expect("setting default subscriber failed");
 
-	let (addr, _handler) = run_server().await?;
-	let url = format!("http://{}", addr);
+	let addr = run_server().await?;
+
+	// WebSocket.
+	{
+		let client = WsClientBuilder::default().build(format!("ws://{}", addr)).await?;
+		let response: String = client.request("say_hello", rpc_params![]).await?;
+		println!("[main]: ws response: {:?}", response);
+		let _response: Result<String, _> = client.request("unknown_method", rpc_params![]).await;
+		let _ = client.request::<String, _>("say_hello", rpc_params![]).await?;
+	}
 
-	let client = HttpClientBuilder::default().build(&url)?;
-	let response: String = client.request("say_hello", rpc_params![]).await?;
-	println!("[main]: response: {:?}", response);
-	let _response: Result<String, _> = client.request("unknown_method", rpc_params![]).await;
-	let _: String = client.request("say_hello", rpc_params![]).await?;
+	tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+
+	// HTTP.
+	{
+		let client = HttpClientBuilder::default().build(format!("http://{}", addr))?;
+		let response: String = client.request("say_hello", rpc_params![]).await?;
+		println!("[main]: http response: {:?}", response);
+		let _response: Result<String, _> = client.request("unknown_method", rpc_params![]).await;
+		let _ = client.request::<String, _>("say_hello", rpc_params![]).await?;
+	}
```

Comment:
Nit: I wonder whether it's worth having a separate `http_and_ws.rs` example to show that you can do this, since this example is, I guess, focused on middleware!

Doesn't need doing in this PR though; just a thought :)

Reply:
Oh actually I see what you're getting at here; middleware works for either http or ws. Let's leave it be :)